### PR TITLE
Added test to verify that design vars, constraints, and objectives

### PR DESCRIPTION
**Pull request by DKilkenny**
https://www.pivotaltracker.com/story/show/170517752

### CODE REVIEW

File: openmdao/drivers/tests/test_pyoptsparse_driver.py [hunk 1]
```diff
@@ -104,6 +106,38 @@ def compute_partials(self, inputs, partials):
...
+
+        model.add_design_var('x', lower=-50.0, upper=50.0)
+        model.add_design_var('y', lower=-50.0, upper=50.0)
+        model.add_objective('f_xy')
+        model.add_constraint('c', upper=-15.0)
+
+        prob.setup()
+        prob.run_driver()
+
+        design_vars = MPI.COMM_WORLD.allgather([prob['x'], prob['y'], prob['c'], prob['f_xy']], root=0)
```
Comment (Kenneth-T-Moore) [inline openmdao/drivers/tests/test_pyoptsparse_driver.py]: Don't need to do an allgather in the test. Just assert that each variable(e.g., prob['x]) equals the static value (e.g., 7.1666667). The test runs on both procs, so such a test would verify that the correct value of the desvars, objs, and cons have been propagated to each proc.

Comment (DKilkenny) [reply openmdao/drivers/tests/test_pyoptsparse_driver.py]: Updated

File: openmdao/drivers/tests/test_pyoptsparse_driver.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import copy
 import sys
 import unittest
+from contextlib import contextmanager
```
Comment (naylor-b) [inline openmdao/drivers/tests/test_pyoptsparse_driver.py]: Don't need this any more.

Comment (DKilkenny) [reply openmdao/drivers/tests/test_pyoptsparse_driver.py]: removed

**Pull request merged**